caffe2/core/typeid.h

```diff
@@ -352,25 +379,41 @@ class TypeMeta {
 // and as a result, we define these two macros slightly differently.
 
 #ifdef _MSC_VER
-#define CAFFE_KNOWN_TYPE(T)                              \
-  template <>                                            \
-  CAFFE2_EXPORT CaffeTypeId TypeMeta::Id<T>() {          \
-    static bool type_id_bit[1];                          \
-    static TypeNameRegisterer<T> registerer(             \
-        reinterpret_cast<CaffeTypeId>(type_id_bit), #T); \
-    return reinterpret_cast<CaffeTypeId>(type_id_bit);   \
+#define CAFFE_KNOWN_TYPE(T)                                                   \
+  template <>                                                                 \
+  CAFFE2_EXPORT TypeId TypeMeta::Id<T>() {                                    \
+    static bool type_id_bit[1];                                               \
+    static const TypeId type_id(reinterpret_cast<intptr_t>(type_id_bit));     \
+    static TypeNameRegisterer<T> registerer(                                  \
+        type_id, #T);                                                         \
+    return type_id;                                                           \
   }
 #else // _MSC_VER
-#define CAFFE_KNOWN_TYPE(T)                              \
-  template <>                                            \
-  CaffeTypeId TypeMeta::Id<T>() {                        \
-    static bool type_id_bit[1];                          \
-    static TypeNameRegisterer<T> registerer(             \
-        reinterpret_cast<CaffeTypeId>(type_id_bit), #T); \
-    return reinterpret_cast<CaffeTypeId>(type_id_bit);   \
+#define CAFFE_KNOWN_TYPE(T)                                                   \
+  template <>                                                                 \
+  TypeId TypeMeta::Id<T>() {                                                  \
+      static bool type_id_bit[1];                                             \
+    static const TypeId type_id(reinterpret_cast<intptr_t>(type_id_bit));     \
+    static TypeNameRegisterer<T> registerer(                                  \
+        type_id, #T);                                                         \
+    return type_id;                                                           \
   }
 #endif
 
-} // namespace caffe2
+// Adapters for legacy code
+using CaffeTypeId = TypeId;
```

Comment:
Continuing the engineering standard upholding here. Let's not keep these adapters everywhere. It should be like a 20 line codemod in fbcode, so it makes no sense to create tech debt here.

Reply:
ok, renamed the new type id class to `CaffeTypeId`, making the adapter obsolete.